lib/IRGen/GenStruct.cpp

```diff
@@ -61,6 +63,31 @@ static StructTypeInfoKind getStructTypeInfoKind(const TypeInfo &type) {
   return (StructTypeInfoKind) type.getSubclassKind();
 }
 
+clang::CXXDestructorDecl *swift::irgen::getCXXDestructor(SILType type) {
+  clang::CXXDestructorDecl *destructor = nullptr;
+  if (!type.getStructOrBoundGenericStruct() ||
+      !type.getStructOrBoundGenericStruct()->getClangDecl())
+    return nullptr;
+  const clang::CXXRecordDecl *cxxRecordDecl = dyn_cast<clang::CXXRecordDecl>(type.getStructOrBoundGenericStruct()->getClangDecl());
+  if (!cxxRecordDecl)
+    return nullptr;
+  for (auto member : cxxRecordDecl->methods()) {
+    if (auto dest = dyn_cast<clang::CXXDestructorDecl>(member)) {
+      destructor = dest;
+      break;
+    }
+  }
+  return destructor;
```

Comment:
```suggestion
  for (auto member : cxxRecordDecl->methods()) {
    if (auto dest = dyn_cast<clang::CXXDestructorDecl>(member)) {
      return dest;
    }
  }
  return nullptr;
```

and remove the `destructor` variable.